uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestMailNotification.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
```

Comment:
This Service has 3 responsibilities: 
- Build the e-mail template
- Build the message (which is the template + from/to addresses)
- Send the e-mail

We should separate these into different classes. What I'm thinking is something like this:
- 3 message template classes (one for each request type), with methods to build the necessary params. These will contain all the template logic that currently is in this class
- A factory to return the right template for a Leave Request instance
- The LeaveRequestMailNotification will represent the item `#2` on the list (it will be the message), and hence it will make sense to have public methods like getRecipients. We should inject template factory here.
- Finally, we'll have a LeaveRequestMailNotificationSender service, which will send the notifications

Since neither the notification nor the template are services, I think we can have a new Email namespace for this: `CRM_HRLeaveAndAbsences_Mail`. Inside it, we can have `CRM_HRLeaveAndAbsences_Mail_Template` for the templates and `CRM_HRLeaveAndAbsences_Mail_Message` for the message class. 

What do you think?